### PR TITLE
More compact defibs: Electric boogaloo

### PR DESCRIPTION
**Pull request by E-MonaRhg**
Not an april fools thing, also
_I am not good with title names._

## What Does This PR Do

This here PR adds a new, advanced compact defibrillator, for the CMO, making their high-risk items more unique. It has a self-charging, basically infinite capacity cell(can't remove it), can revieve through space suits, it is immune to ~~emagging attempts and~~ EMPs. 
The objective to steal the compact defibrillator was changed to steal the advanced compact defibrillator.

Now that the CMO has a better compact defibrillator, the one they had before was moved to RnD.
Also the combat defibrillator for the nukies now has a proper(ish) sprite.

## Why It's Good For The Game

The compact defibrillator was or still is not really all that good of an objective item it is just a small defibrillator. With this PR the CMO will have at least a better, completely unique medical tool.
And the compact defibrillator was moved to RnD. Not really ALL that huge of a balance change, since for example, a paramedic could just get a bag of holding and put the large defibrillator in there. The ability for RnD to make the compact defib just deletes one step, the need to have a bag of holding.

The combat defibrillator was just missing a proper sprite to be honest, now it looks Syndicate-themed.

## Images of changes

Sprites are from tg, but modified a bit.
Image:
image
_(the paramedic does NOT start with a compact defib)_

Animation:
defib

## Changelog
:cl:
add: Added an advanced compact defibrillator and gave it to the CMO.
tweak: Changes the steal objective from stealing the compact defibrillator to stealing the advanced compact defibrillator.
tweak: Makes it possible to make a normal compact defibrillator in RnD.
imageadd: Added sprites for the advanced compact defibrillator and the combat defibrillator.
/:cl:

### CODE REVIEW

**Comment by E-MonaRhg** [comment]
> While I agree with the rest, can you explain why you did this? IMO if a traitor CMO wants to emag his defib, he should be able to and be rewarded with an emagged defib.

Honestly:
1. Early in the making of this PR I just didn't want to make emagged sprites for it.
2. Was thinking it would be too easy for the CMO to get a stun and potentially very lethal and stealthy weapon round-start by just getting an emag, or the same for the traitor that steals it.

If more people think that they should be able to emag/EMP it then I can spend the extra 15-30 minutes making emagged sprites for it.

**Comment by Coolrune206** [comment]
> > it is immune to emagging attempts and EMPs.
> 
> While I agree with the rest, can you explain why you did this? IMO if a traitor CMO wants to emag his defib, he should be able to and be rewarded with an emagged defib.

Eeeehhh... given that the CMO defib works through suits and has an infinite charge? I very much would **not** consider it a good idea to weaponize that for the simple price of an emag/EMP.

**Comment by Wojz** [comment]
> Eeeehhh... given that the CMO defib works through suits and has an infinite charge? I very much would **not** consider it a good idea to weaponize that for the simple price of an emag/EMP.

Why not? It's just a stunprod with a 10% death chance, not the instadeath nukie defib.

File: code/game/objects/items/weapons/defib.dm [hunk 1]
```diff
@@ -217,35 +214,79 @@
...
+/obj/item/defibrillator/compact/combat/loaded/Initialize(mapload)
+	. = ..()
 	cell = new(src)
 	update_icon()
-	return
 
-/obj/item/defibrillator/compact/combat/attackby(obj/item/W, mob/user, params)
+/obj/item/defibrillator/compact/advanced
+	name = "advanced compact defibrillator"
+	desc = "A belt-equipped state-of-the-art defibrillator that can be rapidly deployed in all enviroments. Uses an experimental self-charging cell, meaning that it will (probably) never stop working. Can revive through space suits. Impossible to hack or damage."
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:247]:
```suggestion
	desc = "A belt-mounted state-of-the-art defibrillator that can be rapidly deployed in all environments. Uses an experimental self-charging cell, meaning that it will (probably) never stop working. It does not require thick clothing to be removed before use. It is impossible to hack or damage it."
```
Not sure if "revive" is a bit too OOC, so this tries to adjust that. Also minor spelling tweaks,

File: code/game/objects/items/weapons/defib.dm [hunk 2]
```diff
@@ -217,35 +214,79 @@
...
+		return
+
+/obj/item/defibrillator/compact/advanced/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/bluespace/charging(src)
+	update_icon()
+
+/obj/item/defibrillator/compact/advanced/emag_act(mob/user)
+	if(world.time > next_emag_message)
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but it rejects your attempt... uh oh...")
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:272]:
```suggestion
		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the sequencer, but it rejects your attempt. Uh oh...")
```
Card? Sequencer? Maybe its not that important but I'll suggest it.

File: code/game/objects/items/weapons/defib.dm [hunk 3]
```diff
@@ -217,35 +214,79 @@
...
+		atom_say("Warning: Unauthorised software modification is not advised and might result in the warranty becoming void.")
+		playsound(get_turf(src), 'sound/machines/defib_saftyon.ogg', 50)
+		next_emag_message = world.time + 1 MINUTES
+	else
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but nothing happens.")
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:277]:
```suggestion
		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the sequencer but nothing happens.")
```

File: code/game/objects/items/weapons/defib.dm [hunk 4]
```diff
@@ -585,3 +629,17 @@
...
+/obj/item/twohanded/shockpaddles/syndicate
+	name = "combat defibrillator paddles"
+	desc = "A pair of mounted high-tech paddles with flat plasteel surfaces to revive deceased operatives(unless they exploded). They possess both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
+	icon_state = "syndiepaddles0"
+	item_state = "syndiepaddles0"
+	base_icon_state = "syndiepaddles"
+
+/obj/item/twohanded/shockpaddles/advanced
+	name = "advanced defibrillator paddles"
+	desc = "A pair of mounted high-tech paddles with flat plasteel surfaces that are used to deliver powerful electric shocks. They possess the ability to penetrate armor to deliver shock."
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:642]:
```suggestion
	desc = "A pair of high-tech paddles with flat plasteel surfaces that are used to deliver powerful electric shocks. They possess the ability to penetrate armor to deliver shock."
```

File: code/game/objects/items/weapons/defib.dm [hunk 5]
```diff
@@ -585,3 +629,17 @@
 					playsound(get_turf(src), 'sound/machines/defib_failed.ogg', 50, 0)
 		busy = FALSE
 		update_icon()
+
+/obj/item/twohanded/shockpaddles/syndicate
+	name = "combat defibrillator paddles"
+	desc = "A pair of mounted high-tech paddles with flat plasteel surfaces to revive deceased operatives(unless they exploded). They possess both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:635]:
```suggestion
	desc = "A pair of high-tech paddles with flat plasteel surfaces to revive deceased operatives(unless they exploded). They possess both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
```
These are not mounted onto anything, surely?

File: code/modules/research/designs/medical_designs.dm [hunk 6]
```diff
@@ -72,6 +72,16 @@
 	build_path = /obj/item/defibrillator
 	category = list("Medical")
 
+/datum/design/compact_defib
+	name = "Compact Defibrillator"
+	desc = "A belt-equipped defibrillator that can be rapidly deployed."
```
Comment (S34NW) [inline code/modules/research/designs/medical_designs.dm:77]:
```suggestion
	desc = "A belt-mounted defibrillator that can be rapidly deployed."
```

**Review by S34NW — CHANGES_REQUESTED** [review]
Some suggestions bundled alongside some corrections. Good work on this project, it looks great!

**Review by Kyep — APPROVED** [review]

File: code/game/objects/items/weapons/defib.dm (hunk 2 above)
Comment (E-MonaRhg) [reply code/game/objects/items/weapons/defib.dm]: I think just "card" is descriptive enough, at least to me.

File: code/game/objects/items/weapons/defib.dm (hunk 3 above)
Comment (E-MonaRhg) [reply code/game/objects/items/weapons/defib.dm:277]: Same as above.

File: code/game/objects/items/weapons/defib.dm (hunk 5 above)
Comment (E-MonaRhg) [reply code/game/objects/items/weapons/defib.dm]: Technically mounted to the defibrillator but okay.

File: code/modules/research/designs/medical_designs.dm (hunk 6 above)
Comment (E-MonaRhg) [reply code/modules/research/designs/medical_designs.dm]: Belt-mounted sounds better, yeah.

File: code/game/objects/items/weapons/defib.dm (hunk 1 above)
Comment (E-MonaRhg) [reply code/game/objects/items/weapons/defib.dm]: Changed to this, and revieve to to defibrillate, but:
1. Armor is not descriptive enough
2. Changed to "It is impossible to hack or damage." but without the second "it" at the end.

File: code/game/objects/items/weapons/defib.dm [hunk 7]
```diff
@@ -585,3 +629,17 @@
 					playsound(get_turf(src), 'sound/machines/defib_failed.ogg', 50, 0)
 		busy = FALSE
 		update_icon()
+
+/obj/item/twohanded/shockpaddles/syndicate
+	name = "combat defibrillator paddles"
+	desc = "A pair of high-tech paddles with flat plasteel surfaces to revive deceased operatives(unless they exploded). They possess both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
```
Comment (S34NW) [inline code/game/objects/items/weapons/defib.dm:635]:
```suggestion
	desc = "A pair of high-tech paddles with flat plasteel surfaces to revive deceased operatives (unless they exploded). They possess both the ability to penetrate armor and to deliver powerful or disabling shocks offensively."
```

**Review by S34NW — APPROVED** [review]
looks good, just this once missing space

File: code/game/objects/items/weapons/defib.dm [hunk 8]
```diff
@@ -214,38 +211,82 @@
...
+
+	var/next_emag_message //to prevent spam from the emagging message on the advanced defibrillator
+	var/next_emp_message //same as above, but for EMPs
+
+/obj/item/defibrillator/compact/advanced/attackby(obj/item/W, mob/user, params)
+	if(W == paddles)
+		paddles.unwield()
+		toggle_paddles()
+		update_icon()
+		return
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:263]: `return` here is implied since it's at the end of the proc, so you can get rid of it.
```suggestion
/obj/item/defibrillator/compact/advanced/attackby(obj/item/W, mob/user, params)
	if(W == paddles)
		paddles.unwield()
		toggle_paddles()
		update_icon()
```

File: code/game/objects/items/weapons/defib.dm [hunk 9]
```diff
@@ -214,38 +211,82 @@
...
+
+/obj/item/defibrillator/compact/advanced/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/bluespace/charging(src)
+	update_icon()
+
+/obj/item/defibrillator/compact/advanced/emag_act(mob/user)
+	if(world.time > next_emag_message)
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but it rejects your attempt. Uh oh...")
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:272]: No closing `</span>` here and on the one below.
```suggestion
		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but it rejects your attempt. Uh oh...</span>")
```

File: code/game/objects/items/weapons/defib.dm [hunk 10]
```diff
@@ -214,38 +211,82 @@
...
+		atom_say("Warning: Unauthorised software modification is not advised and might result in the warranty becoming void.")
+		playsound(get_turf(src), 'sound/machines/defib_saftyon.ogg', 50)
+		next_emag_message = world.time + 1 MINUTES
+	else
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but nothing happens.")
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:277]:
```suggestion
		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but nothing happens.</span>")
```

File: code/game/objects/items/weapons/defib.dm [hunk 11]
```diff
@@ -214,38 +211,82 @@
...
+
+/obj/item/defibrillator/compact/advanced/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/bluespace/charging(src)
+	update_icon()
+
+/obj/item/defibrillator/compact/advanced/emag_act(mob/user)
+	if(world.time > next_emag_message)
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but it rejects your attempt. Uh oh...")
+		atom_say("Warning: Unauthorised software modification is not advised and might result in the warranty becoming void.")
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:273]: "May result" instead maybe? Sounds a bit more formal that way.
```suggestion
		atom_say("Warning: Unauthorised software modification is not advised and may result in the warranty becoming void.")
```

File: code/game/objects/items/weapons/defib.dm [hunk 12]
```diff
@@ -214,38 +211,82 @@
...
+		playsound(get_turf(src), 'sound/machines/defib_saftyon.ogg', 50)
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:274]: No need for a `get_turf()` here, can just play the sound on the defib.
```suggestion
		playsound(src, 'sound/machines/defib_saftyon.ogg', 50)
```

File: code/game/objects/items/weapons/defib.dm [hunk 13]
```diff
@@ -214,38 +211,82 @@
...
+	else
+		to_chat(user, "<span class='warning'>You try to silently disable [src]'s safety protocols with the card but nothing happens.")
+
+/obj/item/defibrillator/compact/advanced/emp_act(severity)
+	if(world.time > next_emp_message)
+		atom_say("Warning: Electromagnetic pulse detected. Integrated shielding prevented all potential hardware damage.")
+		playsound(get_turf(src), 'sound/machines/defib_saftyon.ogg', 50)
+		next_emp_message = world.time + 5 SECONDS
+	else
+		return
```
Comment (SabreML) [inline code/game/objects/items/weapons/defib.dm:285]:
```suggestion
/obj/item/defibrillator/compact/advanced/emp_act(severity)
	if(world.time > next_emp_message)
		atom_say("Warning: Electromagnetic pulse detected. Integrated shielding prevented all potential hardware damage.")
		playsound(src, 'sound/machines/defib_saftyon.ogg', 50)
		next_emp_message = world.time + 5 SECONDS
```

File: code/modules/power/cell.dm [hunk 14]
```diff
@@ -284,6 +284,12 @@
 	charge = 0
 	update_icon()
 
+/obj/item/stock_parts/cell/bluespace/charging
+	name = "self-charging bluespace power cell"
+	desc = "An experimental, self-charging, transdimensional power cell."
+	origin_tech =  "powerstorage=10;bluespace=10"
+	self_recharge = 1
```
Comment (SabreML) [inline code/modules/power/cell.dm:291]: Seems to be a boolean.
```suggestion
	self_recharge = TRUE
```

File: code/modules/research/designs/medical_designs.dm [hunk 15]
```diff
@@ -72,6 +72,16 @@
 	build_path = /obj/item/defibrillator
 	category = list("Medical")
 
+/datum/design/compact_defib
+	name = "Compact Defibrillator"
+	desc = "A belt-mounted defibrillator that can be rapidly deployed."
```
Comment (SabreML) [inline code/modules/research/designs/medical_designs.dm:77]: ?
```suggestion
	desc = "A belt-mounted defibrillator for rapid deployment."
```

**Review by SabreML — APPROVED** [review]
Defibn't.
image

File: code/modules/power/cell.dm [hunk 16]
```diff
@@ -293,9 +299,6 @@
 	rating = 6
 	chargerate = 30000
 
-/obj/item/stock_parts/cell/infinite/use()
-	return 1
```
Comment (Fox-McCloud) [inline code/modules/power/cell.dm:297]: Why did you remove this?

Comment (E-MonaRhg) [reply code/modules/power/cell.dm:297]: Quite frankly I don't remember, most likely an accident, fixing now.

**Review by variableundefined — APPROVED** [review]
